src/components/dropdown.js

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import axios from 'axios';
+import SearchHighlight from '../helper/search-highlight';
+import DropdownUser from './dropdown-user';
+import SelectedUsers from './selected-users';
+import '../styles/dropdown.css';
+
+class Dropdown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isDropdownVisible: false,
+      users: [],
+      checkedUsers: [],
+      searchedList: [],
+      selectedUsers: [],
+      searchValue: '',
+    }
+    this.toggleDropdown = this.toggleDropdown.bind(this);
+    this.deleteAllUsers = this.deleteAllUsers.bind(this);
+    this.submitHandler = this.submitHandler.bind(this);
+  }
+
+  componentDidMount() {
+    axios.get('/assets/users_data.json')
+      .then(response => {
+        const data = {}
+        for (const record of response.data) {
+          data[record.id] = record;
+        }
+        this.setState({
+          users: data,
+        });
+      })
+      .catch(function (error) {
+        console.log(error);
+      });
+  }
+
+  toggleDropdown() {
+    this.setState(prevState => ({
+      isDropdownVisible: !prevState.isDropdownVisible,
+      checkedUsers: [],
+    }));
+  }
+
+  getCheckedUsers = event => {
+    let newSelectedUsers = this.state.checkedUsers;
+    let isChecked = event.target.checked;
+    let userId = parseInt(event.target.value);
+    this.setState({
+      checkedUsers: isChecked ? newSelectedUsers.concat(userId) : this.state.checkedUsers.filter(user => user !== userId),
+    });
+  }
+
+  searchUser = (event) => {
+    this.setState({
+      isDropdownVisible: true,
+      searchValue: event.target.value,
+    });
+  }
+
+  deleteUser = (event, userId) => {
+    this.setState({
+      selectedUsers: this.state.selectedUsers.filter(user => user !== userId),
+    });
+  }
+
+  deleteAllUsers() {
+    this.setState({
+      selectedUsers: [],
+    })
+  }
+
+  submitHandler() {
+    const { selectedUsers, checkedUsers } = this.state;
+    this.setState({
+      isDropdownVisible: checkedUsers.length > 0 ? false : true,
+      isSubmitted: true,
+      selectedUsers: [...selectedUsers, ...checkedUsers],
+      checkedUsers: [],
+      searchValue: '',
+    });
+  }
+
+  transformDropdownUsers = () => {
+    const { searchValue, selectedUsers, users } = this.state;
+    let updatedList = Object.values(users).filter(user => !selectedUsers.includes(user.id));
```

Comment:
When iterating through large list always use for loop (either for-in or for-of).